### PR TITLE
Allow disabling .app bundles on Mac

### PR DESCRIPTION
**Pull request by yan12125**
## Description

This PR introduces a CMake option so that users can choose to build keepassxc with traditional UNIX filesystem layouts. Namely, binaries are installed to bin/, libraries to lib/, etc.

## Motivation and context

I want to always use the latest git-master of keepassxc, but I don't want to run ```git pull```, ```make install```, ... manually whenever there's an update. A possible solution is creating a Homebrew formula, and then users can bump to the latest git-master with a simple line:
```
brew reinstall keepassxc
```
(Assume it was installed with ```brew install keepassxc --HEAD```)

However, Homebrew says that it doesn't support application building as FooBar.app well. [1] As a result, I build keepassxc with traditional UNIX filesystem layouts. A sample formula can be found at https://github.com/yan12125/homebrew-staging/blob/master/Formula/keepassxc.rb, which is incomplete yet usable.

[1] https://github.com/Homebrew/brew/pull/1808

## How has this been tested?

First, uninstall KeepassXC.app to avoid possible conflicts.
```
brew tap yan12125/staging
brew install keepassxc --HEAD
keepassxc
```

The built program can open my database. Configurations created by KeepassXC.app are reused. Translations and autotype works, too.

There are two major drawbacks, which seem inevitable for non-bundled applications.
1. There's an extra terminal window if I start keepassxc from the Dock.
2. It has no icon on the Dock

## Screenshots (if appropriate):

Just the same as KeepassXC.app :)

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]** - there's no info about coding style for CMakeLists.txt. If some lines look strange, just tell me.
- ✅ All new and existing tests passed. **[REQUIRED]**
-  I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]** - not applicable on Mac
-  My change requires a change to the documentation and I have updated it accordingly. - Not done yet. Maybe add a note to https://github.com/keepassxreboot/keepassxc/wiki/Building-KeePassXC after this is merged?
-  I have added tests to cover my changes. - sorry but I have no idea to create tests for bulid system changes

### CODE REVIEW

**Comment by TheZ3ro** [comment]
@louib @weslly can you guys test this on Mac? Thanks

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,7 @@ option(WITH_GUI_TESTS "Enable building of GUI tests" OFF)
 option(WITH_DEV_BUILD "Use only for development. Disables/warns about deprecated methods." OFF)
 option(WITH_ASAN "Enable address sanitizer checks (Linux only)" OFF)
 option(WITH_COVERAGE "Use to build with coverage tests (GCC only)." OFF)
+option(WITH_APP_BUNDLE "Enable Application Bundle for macOS" ON)
```
Comment (phoerious) [inline CMakeLists.txt]: OS X

Comment (yan12125) [inline CMakeLists.txt]: Updated :)

Comment (weslly) [inline CMakeLists.txt]: [Isn't it macOS now?](https://www.engadget.com/2016/06/13/os-x-is-now-macos/)

Comment (phoerious) [inline CMakeLists.txt]: Is it? I didn't know.

Comment (yan12125) [inline CMakeLists.txt]: Just checked it - most documents/comments in KeepassXC use "OS X". For me it's fine to use the new term or the old term for consistency with the rest of KeepassXC.

Comment (phoerious) [inline CMakeLists.txt]: We normalized our terminology to OS X, but I'm fine with changing it, as long as we do it consistently.

**Comment by yan12125** [comment]
Thanks!

**Pull request merged**